exercises/tom_hampson/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Skeleton main routine
+ */
+#include "Square.h"
+#include "Circle.h"
+#include "Triangle.h"
+#include "ShapeSorter.h"
+
+int main(int, char **)
+{
+  //Square square(5);
+  //square.print();
+
+  //Triangle triangle(3, 5);
+  //triangle.print();
+
+  //Circle circle(4);
+  //circle.print();
+
+  //Rectangle rectangle(2, 4);
+  //rectangle.print();
+
+  ShapeSorter sorter;
+  sorter.addShape(std::make_shared<Circle>(3));
+  sorter.addShape(std::make_shared<Square>(5));
+  sorter.addShape(std::make_shared<Triangle>(5,4));
+  sorter.addShape(std::make_shared<Triangle>(6,4));
```

Comment:
Would also have been worth considering a unique_ptr here, as we may not need the shared ownership of each shape, which comes at a cost. But not worth changing.

Reply:
Noted